ring-core/src/ring/util/response.clj

```diff
@@ -195,20 +216,36 @@
              dir-entry  (.getEntry jar-file (add-ending-slash entry-name))]
          (and dir-entry (.isDirectory dir-entry)))))
 
+(defmethod resource-info :jar
+    [^java.net.URL url]
+    (let [conn (.openConnection url)]
+      {:content        (delay (.getInputStream conn))
```

Comment:
Since `resource-info` is ultimately being used to construct a response map, the content-length and last-modified fields will always be used if they're available. If people want to use `resource-info` in some other way, that's fine, but it's purpose should be to be used to construct as complete a response map as possible.